src/components/AddressCard.tsx

```diff
@@ -81,81 +95,147 @@ const AddressCard = ({ style, addressHash, onSettingsPress }: AddressCardProps)
     })
   }
 
+  const handleDefaultAddressToggle = async () => {
+    setLoading(true)
+
+    try {
+      const newSettings = { ...address.settings, isDefault: true }
+
+      await persistAddressSettings({ ...address, settings: newSettings })
+      dispatch(addressSettingsSaved({ ...address, settings: newSettings }))
+
+      showToast('This is now the default address')
+
+      posthog?.capture('Address: Used address card default toggle')
+    } catch (e) {
+      console.error(e)
+
+      posthog?.capture('Error', { message: 'Could not use address card default toggle' })
+    }
+    setLoading(false)
+  }
+
   return (
-    <LinearGradient style={style} colors={[bgColor, colord(bgColor).darken(0.1).toHex()]} start={{ x: 0.1, y: 0.3 }}>
-      <Header>
-        <AddressBadgeContainer>
-          {address.settings.isDefault && <DefaultAddressBadge size={18} color={textColor} />}
-          <AddressBadgeStyled
-            addressHash={address.hash}
-            hideSymbol
+    <View
+      style={[
+        style,
+        {
+          shadowColor: 'black',
+          shadowOffset: { height: 5, width: 0 },
+          shadowOpacity: theme.name === 'dark' ? 0.5 : 0.15,
+          shadowRadius: 8,
+          elevation: 10,
+          borderColor: outterBorderColor
+        }
+      ]}
+    >
+      <CardGradientContainer
+        colors={[bgColor, colord(bgColor).saturate(0.15).lighten(0.05).toHex()]}
+        start={{ x: 0.1, y: 0.3 }}
+      >
+        <Header>
+          <AddressBadgeContainer>
+            <AddressBadgeStyled
+              addressHash={address.hash}
+              hideSymbol
+              color={textColor}
+              textStyle={{
+                fontSize: 23,
+                fontWeight: '700'
+              }}
+              showCopyBtn
+            />
+          </AddressBadgeContainer>
+          <HeaderButtons>
+            <Button
+              onPress={handleDefaultAddressToggle}
+              customIcon={
+                <DefaultAddressBadge
+                  strokeOnly={!isDefaultAddress}
+                  size={18}
+                  color={isDefaultAddress ? theme.global.accent : textColor}
+                />
+              }
+              style={{ backgroundColor: isDefaultAddress ? 'rgba(255, 255, 255, 0.2)' : buttonsBackground }}
+              round
+            />
```

Comment:
If the address is already the default one, we don't need to call the `handleDefaultAddressToggle`. It makes an unnecessary call to the store, displays the spinner modal, and the toast. I think it should only do all that when changing the default address.